### DISCUSSION TITLE
Video Transcoding

### DISCUSSION THREAD

**Discussion by shanehandley** [supabase/supabase#18002] [Feature Requests]
It would be great if Supabase storage provided video optimisation in the same way that it provides image optimisation.

Service such as [cloudinary](https://cloudinary.com/documentation/video_optimization) offer this functionality, but it's suboptimal as a user because they often require that their service handles the file storage - as a supabase storage user, this means I have to split image (supabase) and video (other) uploads across services.

Having both in supabase storage, and being able to transform both types of media, would be optimal.

**Comment by gblikas** [comment]
@shanehandley major bump. This discussion might be related #1407. It seems like image was a simple ask, but video optimization might be complex.

If Supabase decides not to do this, it would be nice to at least have a [Supabase Functions](https://supabase.com/docs/guides/functions) example showing how to "hotwire" the functionality. 

**edit** It looks like this might be a [limitation on the Edge functionality](https://supabase.com/docs/guides/functions/limits) itself: 

> ## Runtime limits
> - Maximum Memory: 256MB
> - Maximum Duration (Wall clock limit): This is the duration an Edge Function worker will stay active. During this period, a worker can serve multiple requests or process background tasks.
> - Free plan: 150s
> - Paid plans: 400s
> - Maximum CPU Time: 2s (Amount of actual time spent on the CPU per request - does not include async I/O.)
> - Request idle timeout: 150s (If an Edge Function doesn't send a response before the timeout, 504 Gateway Timeout will be returned)

I assume for anything video related, this would be too limiting to implement as an actual solution.
